datafusion/src/logical_plan/mod.rs

```diff
@@ -27,7 +27,7 @@ mod display;
 mod expr;
 mod extension;
 mod operators;
-pub mod plan;
+mod plan;
```

Comment:
I think this is not backward compatible

Reply:
I agree with @Jimexist 's implicit suggestion that we should leave the code as `pub mod plan;`

Reply:
@Jimexist 
@alamb 
This comment has been addressed

Reply:
@Jimexist @alamb 
I have doubts about this backward compatible.
Before this commit #1290, this is `mod plan`, and this commit is not in the release 6.0.0 datafusion.
The reason i did this is to keep code style with original usage:
```
pub use plan::{
    JoinConstraint, JoinType, LogicalPlan, Partitioning, PlanType, PlanVisitor,
};
```
and make some struct public we wanted.